orangecontrib/spectroscopy/io/perkinelmer.py

```diff
@@ -0,0 +1,471 @@
+import struct
+import os
+
+import numpy as np
+
+from Orange.data import Table, Domain, FileFormat
+
+from orangecontrib.spectroscopy.io.util import SpectralFileFormat, _spectra_from_image_2d
+
+
+### This file is based on the development chain indicated below:
+
+## The below file readers are based on file readers written by Specio
+## (https://github.com/paris-saclay-cds/specio) with the following
+## license:

```

Comment:
I guess it's the `BlockReader` class that is borrowed from `specio`? Let's put that class in `orangecontrib.spectroscopy.utils.specio` with the other code-copies.

Reply:
I made a new file to store this part of the code. I think the guys from Diamond refactored the specio code a bit.

Reply:
Yes, I saw that after I made this comment.

Great, can you put the `specio` file in `orangecontrib/spectroscopy/utils/specio`? That's where the other code copies are (pymca, etc)